.ci/scripts/generate_build_table.py

```diff
@@ -31,6 +31,7 @@
                     if "withModule" in doc["stages"][stage]:
                         withModule = doc["stages"][stage]["withModule"]
                     if "when" in doc["stages"][stage]:
-                        when = f"optional"
+                        if "not_changeset_full_match" not in doc["stages"][stage]["when"]:
+                            when = f"optional"
```

Comment:
```suggestion
                            when = "optional"
```
Nit: strictly speaking, it's a PEP8 violation to use a f-string if there is no interpolation.